staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -45,32 +45,34 @@ func ValidateAuthenticationConfiguration(c *api.AuthenticationConfiguration) fie
 	root := field.NewPath("jwt")
 	var allErrs field.ErrorList
 
-	// This stricter validation is solely based on what the current implementation supports.
-	// TODO(aramase): when StructuredAuthenticationConfiguration feature gate is added and wired up,
-	// relax this check to allow 0 authenticators. This will allow us to support the case where
-	// API server is initially configured with no authenticators and then authenticators are added
-	// later via dynamic config.
-	if len(c.JWT) == 0 {
-		allErrs = append(allErrs, field.Required(root, fmt.Sprintf(atLeastOneRequiredErrFmt, root)))
-		return allErrs
-	}
+	// We allow 0 authenticators in the authentication configuration.
+	// This allows us to support scenarios where the API server is initially set up without
+	// any authenticators and then authenticators are added later via dynamic config.
 
-	// This stricter validation is because the --oidc-* flag option is singular.
-	// TODO(aramase): when StructuredAuthenticationConfiguration feature gate is added and wired up,
-	// remove the 1 authenticator limit check and add set the limit to 64.
-	if len(c.JWT) > 1 {
-		allErrs = append(allErrs, field.TooMany(root, len(c.JWT), 1))
+	if len(c.JWT) > 64 {
+		allErrs = append(allErrs, field.TooMany(root, len(c.JWT), 64))
 		return allErrs
 	}
 
-	// TODO(aramase): right now we only support a single JWT authenticator as
-	// this is wired to the --oidc-* flags. When StructuredAuthenticationConfiguration
-	// feature gate is added and wired up, we will remove the 1 authenticator limit
-	// check and add validation for duplicate issuers.
+	seenIssuers := sets.NewString()
+	seenDiscoveryURLs := sets.NewString()
 	for i, a := range c.JWT {
 		fldPath := root.Index(i)
 		_, errs := validateJWTAuthenticator(a, fldPath, utilfeature.DefaultFeatureGate.Enabled(features.StructuredAuthenticationConfiguration))
 		allErrs = append(allErrs, errs...)
+
+		if seenIssuers.Has(a.Issuer.URL) {
+			allErrs = append(allErrs, field.Duplicate(fldPath.Child("issuer").Child("url"), a.Issuer.URL))
+		}
+		seenIssuers.Insert(a.Issuer.URL)
+
+		if len(a.Issuer.DiscoveryURL) > 0 {
+			discoveryURL := strings.TrimRight(a.Issuer.DiscoveryURL, "/")
```

Comment:
no longer need to trim this, I think

Reply:
Isn't `GET https://example.com` and `GET https://example.com/` lead to the same request being made?  I know we cannot trim the issuer URL since that is suppose to be based on pure string comparisons, but discovery URL is purely for making network requests?

Reply:
Given that we receive the complete URL from the user for discovery, I think it's still necessary to trim it. This is because `https://example.com/.well-known/openid-configuration` and `https://example.com/.well-known/openid-configuration/` are identical, and we want to ensure they're not treated as distinct URL.

Reply:
Ah, @enj beat me to it 😄

Reply:
> Isn't GET https://example.com and GET https://example.com/ lead to the same request being made?

If the discoveryURL path is exactly `/`, yes, but otherwise, no.

https://go.dev/play/p/V3Ajj1uAMmG

```
$curl https://accounts.google.com/.well-known/openid-configuration

{
 "issuer": "https://accounts.google.com",
...
```

```
$ curl https://accounts.google.com/.well-known/openid-configuration/

<html lang=en><meta charset=utf-8><meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width"><title>Error 404 (Not Found)!!1...
```

Reply:
Hmm yeah I guess you can treat those as differently on the server side.  Well I suppose that makes this check a bit simpler since we don't need to trim.

Reply:
Makes sense! I was trying with `login.microsoftonline.com` which works.

```
➜ curl https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration

{
 "issuer": "issuer": "https://login.microsoftonline.com/{tenantid}/v2.0",
...
```
```
➜ curl https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration/

{
 "issuer": "issuer": "https://login.microsoftonline.com/{tenantid}/v2.0",
...
```